src/scripts/main.js

```diff
@@ -1,11 +1,26 @@
 'use strict';
 
 function waitFor(element, eventName) {
-  // write your code here
+  return new Promise((resolve) => {
+    const handler = () => {
+      const message
```

Comment:
The variable 'message' is declared but not immediately assigned. It's better to declare and assign it at the same time for readability.